components/world-4/breeding/pets.tsx

```diff
@@ -0,0 +1,522 @@
+    }, [appContext]);
+
+    // our sort options are fixed, so just statically set them.
+    const sortOptions = ["Level", "Least Time to Next Level"];
+
+    const petsToShow = useMemo(() => {
+        // If we are still loading, do nothing.
+        if (!breeding) {
+            return [];
+        }
+        // Start with a base list of all pets
+        let pets: Pet[] = breeding.basePets.filter(pet => pet.data.petId != "_");
+
+        // Now we sort
+        return pets?.sort((pet1, pet2) => {
+            // Base case scenario they are just by index.
+            const indexSort = pet1.index > pet2.index ? 1 : -1;
+
+            // Least time to next level sort function
+            function sortByTime(pet1: Pet, pet2: Pet) {
+                // If pet is max level, move it to the end
+                if (pet1.breedingLevel >= 10) {
+                    if (pet2.breedingLevel >= 10) {
+                        return indexSort;
+                    } else {
+                        return 1;
+                    }
+                }
+                // If pet is max level, move it to the end
+                if (pet2.breedingLevel >= 10) {
+                    return -1;
+                }
+                // Else sort by time till next level.
+                return (pet1.getNextBreedingGoal() - pet1.breedingProgress) > (pet2.getNextBreedingGoal() - pet2.breedingProgress) ? 1 : -1;
+            }
+
+            // Sort by level sort function
+            function sortByLevel(pet1: Pet, pet2: Pet) {
+                if (pet1.breedingLevel == pet2.breedingLevel) {
+                    // If level is equal, sort by time until next level
+                    return sortByTime(pet1, pet2);
+                } else {
+                    // if level isn't equal, just sort by level
+                    return pet1.breedingLevel > pet2.breedingLevel ? -1 : 1;
+                }
+            }
+
+            switch (sort) {
+                case "Level":
+                    return sortByLevel(pet1, pet2);
+                case "Least Time to Next Level":
+                    return sortByTime(pet1, pet2);
+                default:
+                    return indexSort;
+            }
+        })
+    }, [breeding, sort])
+
+    if (!breeding) {
+        return (
+            <Box>
+                Still loading
+            </Box>
+        )
+    }
+
+    return (
+        <Box gap="small">
+            <Text>Breedability</Text>
+            <Box direction="row" gap="medium" margin={{bottom: "small"}}>
+                <Select size="small"
+                    placeholder="Sort by"
+                    clear
+                    value={sort}
+                    options={sortOptions}
+                    onChange={({ value: nextValue }) => { setSort(nextValue); }}
+                />
+            </Box>
+            <Text size='medium'>Breedability speed multiplier : {nFormatter(breeding.getBreedingSpeed(starSignEquipped, silkRodeChip))}</Text>
```

Comment:
Please use `TextAndLabel` component for this, also potentially missing a `x` in the end? i.e. it's like `4.81x`

Reply:
Done